### PR TITLE
feat: set labels to predictions when clicking use labels

### PR DESCRIPTION
**Pull request by Sparkier**
fix EOT-94

### CODE REVIEW

File: src/lib/components/Instances/InstanceTable.svelte [hunk 1]
```diff
@@ -66,6 +67,29 @@
...
+		);
+		project.metric_name = projectRes.metric_name;
+		// update instances with labels = predictions
+		instances = await Promise.all(
+			instances.map(async (instance) => {
+				const prediction = await predictions[instance.id];
+				if (prediction === null || instance.label !== null) return instance;
+				const newInstance = { ...instance, label: prediction };
+				await updateInstance(newInstance);
+				console.log(newInstance);
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTable.svelte:88]: nit: log.

File: src/lib/components/Instances/InstanceTable.svelte [hunk 2]
```diff
@@ -66,6 +67,29 @@
...
+		project.show_labels = true;
+		const projectRes = await toggleProjectLabels(
+			project.id,
+			prompt,
+			project.show_labels,
+			project.metric_name
+		);
+		project.metric_name = projectRes.metric_name;
+		// update instances with labels = predictions
+		instances = await Promise.all(
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTable.svelte:82]: should we instead do this with a backend endpoint? This is a ton of fetch requests and a bit complicated. Instead, we can have one backend endpoint that does this in one operation maybe?

**Review by cabreraalex — CHANGES_REQUESTED** [review]

**Comment by Sparkier** [comment]
@cabreraalex do you see the second commit here? It shows in the branch but somehow not in the PR for me...

**Comment by cabreraalex** [comment]
only see one commit?

File: src/lib/components/Instances/InstanceTable.svelte [hunk 3]
```diff
@@ -175,7 +185,7 @@
 				</tr>
 			</thead>
 			<tbody>
-				{#each instances as instance, i (instance.id)}
+				{#each instances as instance, i}
```
Comment (cabreraalex) [inline src/lib/components/Instances/InstanceTable.svelte:188]: why remove?

**Comment by Sparkier** [comment]
> only see one commit?

I think this is a GitHub bug, can you click on edit, select the same base branch, and click update base? That fixed it for me.

File: src/lib/api.ts [hunk 4]
```diff
@@ -207,3 +207,13 @@ export async function changeProjectMetric(projectId: string, metric: string | nu
 		body: JSON.stringify({ metric })
 	});
 }
+
+export async function updateInstances(instances: Tables<'instances'>[]) {
```
Comment (cabreraalex) [inline src/lib/api.ts:211]: better function name? this just updates predictions?

File: src/routes/api/instances/labels/predictions/+server.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import type { Tables } from '$lib/supabase.js';
+import { error } from '@sveltejs/kit';
+
+export async function POST({ locals: { supabase }, request }) {
```
Comment (cabreraalex) [inline src/routes/api/instances/labels/predictions/+server.ts:4]: It's early but I'm confused - this is just updating instances, not predictions? why is it in predictions path? it updates labels right?

File: src/lib/api.ts (hunk 4 above)
Comment (cabreraalex) [reply src/lib/api.ts:211]: I think I'm confused by the fetch path actually

**Review by cabreraalex — APPROVED** [review]

File: src/lib/api.ts (hunk 4 above)
Comment (Sparkier) [reply src/lib/api.ts:211]: aha yeah good catch

**Pull request merged**